### DISCUSSION TITLE
How can I get the party/attorney information using the API?

### DISCUSSION THREAD

**Discussion by mlissner** [freelawproject/courtlistener#3746] [Q&A]
I'm using the search API to identify a docket of interest. Once I do, how do I access parties and their attorneys' information in a manner similar to what is displayed in the Parties and Attorneys Tab on the front-end?

**Comment by mlissner** [comment]
The case I use for most of my demoing purposes is the PACER fees class action lawsuit:

https://www.courtlistener.com/docket/4214664/parties/national-veterans-legal-services-program-v-united-states/

It has a docket ID of `4214664`.

Using that, I can go to the parties API and filter to just that docket:

```
GET https://www.courtlistener.com/api/rest/v3/parties/?docket=4214664
```

That returns: 

```
{
    "count": 34,
    "next": "https://www.courtlistener.com/api/rest/v3/parties/?docket=4214664&page=2",
    "previous": null,
    "results": [
        {
            "resource_uri": "https://www.courtlistener.com/api/rest/v3/parties/12911884/",
            "id": 12911884,
            "attorneys": [
                {
                    "role": 1,
                    "docket": "https://www.courtlistener.com/api/rest/v3/dockets/4214664/",
                    "attorney": "https://www.courtlistener.com/api/rest/v3/attorneys/8648984/",
                    "date_action": null
                },
                {
                    "role": 2,
                    "docket": "https://www.courtlistener.com/api/rest/v3/dockets/4214664/",
                    "attorney": "https://www.courtlistener.com/api/rest/v3/attorneys/8648984/",
                    "date_action": null
                }
            ],
            "party_types": [
                {
                    "docket": "https://www.courtlistener.com/api/rest/v3/dockets/4214664/",
                    "name": "Interested Party",
                    "date_terminated": null,
                    "extra_info": "",
                    "highest_offense_level_opening": "",
                    "highest_offense_level_terminated": "",
                    "criminal_counts": [],
                    "criminal_complaints": []
                }
            ],
            "date_created": "2023-08-30T11:14:41.656029-07:00",
            "date_modified": "2023-08-30T11:14:41.656037-07:00",
            "name": "TROY LAW, PLLC",
            "extra_info": ""
        },
...
```

And that's your answer, but you need to make sense of it. You'll see towards the end that this is for `TROY LAW`, so let's go look at the HTML for that in CL:

![image](https://github.com/freelawproject/courtlistener/assets/236970/06bd8da2-f88b-4a0e-bd3a-6d9b199778ab)

<sub>Ignore the pencil icons, those are b/c I'm an admin</sub>

This one is a bit complicated because Troy Law is representing its own interests in this case, but here's how I think of it:

1. The name of the party is: `"name": "TROY LAW, PLLC"`
2. That party has a party type of `"name": "Interested Party",`
3. There are two values in the `attorneys` key. This key is a bit poorly named, but the idea is that the attorney in the key has two roles for the party in this docket (role `1` and role `2`). These correspond to Attorney to be Noticed and Lead Attorney you see in the screenshot.
4. If you want the attorney's information, click the link in the attorney section. That gets you:

    ```
    {
        "resource_uri": "https://www.courtlistener.com/api/rest/v3/attorneys/8648984/",
        "id": 8648984,
        "parties_represented": [
            {
                "role": 1,
                "docket": "https://www.courtlistener.com/api/rest/v3/dockets/4214664/",
                "party": "https://www.courtlistener.com/api/rest/v3/parties/12911884/",
                "date_action": null
            },
            {
                "role": 2,
                "docket": "https://www.courtlistener.com/api/rest/v3/dockets/4214664/",
                "party": "https://www.courtlistener.com/api/rest/v3/parties/12911884/",
                "date_action": null
            }
        ],
        "date_created": "2023-08-30T11:14:41.893765-07:00",
        "date_modified": "2024-02-05T10:11:47.090192-08:00",
        "name": "John Troy",
        "contact_raw": "TROY LAW, PLLC\n41-25 Kissena Boulevard, Suite 110\nFlushing, NY 11355\n718-762-2332\nEmail: [johntroy@troypllc.com\n](mailto:johntroy@troypllc.com\n)",
        "phone": "(718) 762-2332",
        "fax": "",
        "email": "[johntroy@troypllc.com](mailto:johntroy@troypllc.com)"
    }
    ```

And that's the full information you need for that party. Iterate over the remaining 33, and you should be in business.

**Reply by mlissner** [reply]
Note that as of API v4.1, there's a new feature to filter the nested results of the Party endpoint. As [the docs](https://www.courtlistener.com/help/api/rest/pacer/#party-endpoint) say: 

![image](https://github.com/user-attachments/assets/05075827-053d-4259-bc56-869ec2d92ad3)

So it's probably worth adding `filter_nested_results=true` to your API requests above.

**Comment by sanazyavari** [comment]
Where can I find the definitions for roles (1 or 2 ..)? Thanks

**Reply by mlissner** [reply]
Thanks for asking. If you do an HTTP options request to the endpoint (or press the OPTIONS) button on the browsable API, it'll give you the details. The idea is that the API is full self-describing.

![image](https://github.com/freelawproject/courtlistener/assets/236970/55e8e7f0-8b70-4464-a135-edf6d631d006)

**Comment by legaltextai** [comment]
it says i don't have access to parties info even though i am logged in 
<img width="952" alt="Screenshot 2025-05-26 at 7 55 00 AM" src="https://github.com/user-attachments/assets/58e7826f-1d60-4441-9efb-484c9cedf8d9" />

**Reply by mlissner** [reply]
Grab me on Slack and we can get you sorted!

**Comment by ledwindra** [comment]
Hi,

Is it also possible to get the party/attorney information using the bulk data? I don't seem to find the any table corresponding to party/attorney through this link: https://com-courtlistener-storage.s3-us-west-2.amazonaws.com/list.html?prefix=bulk-data/. From the schema diagram, it looks the the information is in the `people_db_partytype`.

This can be helpful for my use case because the `plaintiff` and `defendant` columns from the `dockets-2025-04-30.csv.bz2` table doesn't seem to be the complete parties. For example, docket `id == 6108804` has `plaintiff == CLARKE` and `defendant == MICHIGAN DEPARTMENT OF , ET AL`. When I look the REST API (https://www.courtlistener.com/api/rest/v4/dockets/6108804/), I found the `absolute_url` == [/docket/6108804/clarke-v-michigan-department-of-corrections/](https://www.courtlistener.com/docket/6108804/parties/clarke-v-michigan-department-of-corrections/) and there are more defandants (thus et al in the table and REST API.

I've searched other Q&A but it looks like all refers to API, not bulk data:

<img width="1311" alt="image" src="https://github.com/user-attachments/assets/2fe72cb4-6ee7-445f-bc34-79e0c64d0e82" />

The discussion titled [Fetching documents in bulk from RECAP?](https://github.com/freelawproject/courtlistener/discussions/2725) doesn't talk about getting the information apparently.

I also don't have the access to the parties REST API:

<img width="1173" alt="image" src="https://github.com/user-attachments/assets/64c135da-f88f-4918-87b7-124bb6add912" />

Thank you.

Best,
Lukman

**Reply by mlissner** [reply]
Would the `case_name` field in the dockets data do it for you? It'll have the information that shows up in the URL. If you want the *full* party information, you'll need to crawl the party API, but many cases don't have the information, and you won't be able to crawl all 60M items.

**Reply by ledwindra** [reply]
Thank you for the answer. The `case_name` field wouldn't do it for me. In my case, the *full* party information is necessary, even though many cases don't have the information. But I understand if that's the policies that I won't be able to crawl all 60M items.
